lib/async/redis/context/psubscribe.rb

```diff
@@ -0,0 +1,36 @@
+require_relative 'generic'
+
+module Async
+  module Redis
+    module Context
+      class Psubscribe < Generic
+        def initialize(pool, channels)
+          super(pool)
+
+          @channels = channels
+
+          psubscribe(channels)
+        end
+
+        def listen
+          return @connection.read_response
+        end
+
+        private
+
+        def psubscribe(channels)
+          @connection.write_request ['PSUBSCRIBE', *channels]
+          @connection.flush
+
+          response = nil
+
+          channels.length.times do |i|
```

Comment:
```suggestion
          channels.length.times do
```

Would this do the same thing?